### PR TITLE
Reduce timestamp granularity

### PR DESCRIPTION
**Pull request by srinivas81**
@laktech This code is built on top of the pull-request #134 with your code review comments. I talked to [pmerlet-at-didomi] and he wasn't able to finish the changes requested, so I took it over. Thanks.

### CODE REVIEW

**Comment by srinivas81** [comment]
@laktech Any update on this?

**Review by laktech — CHANGES_REQUESTED** [review]

File: iabtcf-encoder/src/main/java/com/iabtcf/encoder/TCStringEncoder.java [hunk 1]
```diff
@@ -185,8 +186,10 @@ private TCStringEncoderV2(TCStringEncoder.Builder builder)
             }
 
             version = checkBounds(builder.version, FieldDefs.CORE_CMP_VERSION);
-            created = Objects.requireNonNull(builder.created);
-            updated = Objects.requireNonNull(builder.updated);
+            /* We are using the value of updated field in the builder to populate the created in order to keep the
+               values of created and update same as per the new TCF specifications */
+            created = Objects.requireNonNull(builder.updated.truncatedTo(ChronoUnit.DAYS));
```
[the remark this answers is not in the document]
Comment (srinivas81) [reply iabtcf-encoder/src/main/java/com/iabtcf/encoder/TCStringEncoder.java:191]: I did this on purpose and I also changed the unit tests accordingly. If you look at the last sentence of the [spec](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#what-happened-to-created-and-lastupdated), you will see that IAB now wants the same value in both created and updated.

File: iabtcf-encoder/src/test/java/com/iabtcf/encoder/utils/TestUtils.java [hunk 2]
```diff
@@ -24,8 +24,9 @@
 
 public class TestUtils {
 
+    private static final long DAY_AS_DECISECONDS = 24 * 60 * 60 * 10;
```
[the remark this answers is not in the document]
Comment (srinivas81) [reply iabtcf-encoder/src/test/java/com/iabtcf/encoder/utils/TestUtils.java:27]: Nope, I will remove it. I will wait till I hear from you regarding the previous comment.

File: iabtcf-encoder/src/main/java/com/iabtcf/encoder/TCStringEncoder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (laktech) [reply iabtcf-encoder/src/main/java/com/iabtcf/encoder/TCStringEncoder.java:191]: cool thanks for clarifying

**Comment by laktech** [comment]
looks good thx

**Comment by srinivas81** [comment]
> looks good thx

@laktech I resolved the other comment. If you are good, please approve the pull request. Thanks.

File: iabtcf-decoder/src/main/java/com/iabtcf/decoder/TCStringV2.java [hunk 3]
```diff
@@ -263,15 +264,15 @@ public Instant getCreated() {
         if (cache.add(CORE_CREATED)) {
             consentRecordCreated = Instant.ofEpochMilli(bbv.readBits36(CORE_CREATED) * 100);
         }
-        return consentRecordCreated;
+        return consentRecordCreated.truncatedTo(ChronoUnit.DAYS);
```
Comment (laktech) [inline iabtcf-decoder/src/main/java/com/iabtcf/decoder/TCStringV2.java:267]: i'm minus one on this. we should return whatever was encoded.

**Review by laktech — CHANGES_REQUESTED** [review]

File: iabtcf-decoder/src/main/java/com/iabtcf/decoder/TCStringV2.java (hunk 3 above)
Comment (srinivas81) [reply iabtcf-decoder/src/main/java/com/iabtcf/decoder/TCStringV2.java:267]: Yeah, I was not sure myself. I think it's better to return whatever is encoded. I will make the change.

Comment (srinivas81) [reply iabtcf-decoder/src/main/java/com/iabtcf/decoder/TCStringV2.java]: @laktech Ok done pushing my changes.

File: iabtcf-encoder/src/test/java/com/iabtcf/encoder/BitWriterTest.java [hunk 4]
```diff
@@ -336,7 +334,7 @@ public void testSimpleEncode() {
         BitWriter bw = new BitWriter();
         bw.write(2, 6);
         bw.write(Instant.parse("2020-01-26T17:01:00Z").toEpochMilli() / 100, 36);
-        bw.writeDays(Instant.parse("2021-02-02T17:01:00Z"), FieldDefs.CORE_CREATED);
+        bw.write(Instant.parse("2021-02-02T17:01:00Z"), FieldDefs.CORE_LAST_UPDATED);
```
Comment (laktech) [inline iabtcf-encoder/src/test/java/com/iabtcf/encoder/BitWriterTest.java:337]: hmm.. why was this changed from CORE_CREATED to LAST_UPDATED field?

Comment (laktech) [reply iabtcf-encoder/src/test/java/com/iabtcf/encoder/BitWriterTest.java:337]: I think this is correct..

        bw.write(2, 6);  => version field
        bw.write(Instant.parse("2020-01-26T17:01:00Z").toEpochMilli() / 100, 36); => created field
        bw.write(Instant.parse("2021-02-02T17:01:00Z"), FieldDefs.CORE_LAST_UPDATED); => updated field.

**Review by laktech — APPROVED** [review]

**Comment by laktech** [comment]
nice work, looks good

**Review by iabmayank — APPROVED** [review]
